cluster-operator/src/main/java/io/strimzi/operator/cluster/model/cruisecontrol/ApiCredentials.java

```diff
@@ -0,0 +1,245 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.cluster.model.cruisecontrol;
+
+import io.fabric8.kubernetes.api.model.Secret;
+import io.strimzi.operator.cluster.model.CruiseControl;
+import io.strimzi.operator.common.InvalidConfigurationException;
+import io.strimzi.operator.common.Util;
+import io.strimzi.operator.common.model.PasswordGenerator;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static io.strimzi.operator.common.model.cruisecontrol.CruiseControlApiProperties.AUTH_FILE_KEY;
+import static io.strimzi.operator.common.model.cruisecontrol.CruiseControlApiProperties.HEALTHCHECK_PASSWORD_KEY;
+import static io.strimzi.operator.common.model.cruisecontrol.CruiseControlApiProperties.HEALTHCHECK_USERNAME;
+import static io.strimzi.operator.common.model.cruisecontrol.CruiseControlApiProperties.REBALANCE_OPERATOR_PASSWORD_KEY;
+import static io.strimzi.operator.common.model.cruisecontrol.CruiseControlApiProperties.REBALANCE_OPERATOR_USERNAME;
+import static io.strimzi.operator.common.model.cruisecontrol.CruiseControlApiProperties.TOPIC_OPERATOR_PASSWORD_KEY;
+import static io.strimzi.operator.common.model.cruisecontrol.CruiseControlApiProperties.TOPIC_OPERATOR_USERNAME;
+import static io.strimzi.operator.common.model.cruisecontrol.CruiseControlApiProperties.TOPIC_OPERATOR_USERNAME_KEY;
+
+/**
+ * Uses information in a Kafka Custom Resource to generate a API credentials configuration file to be used for
+ * authenticating to Cruise Control's REST API.
+ */
+public class ApiCredentials {
```

Comment:
Have you considered wrapping this into the `CruiseControl` class? Similarly to how for example the `JmxModel` class is used in `KafkaCluster`? I think it might provide better isolation between the classes and keep a single interface between the CruiseControlReconciler and the model classes.

Reply:
I hadn't until you mentioned it TBH.  Just refactored the classes model classes a little bit so that `ApiCredentials` is not referenced in `CruiseControlReconciler`. This should provide better isolation between the Reconciler and model classes than before. Let me know if I misinterpreted the ask or you want me to takes this further

Reply:
This is good @kyguy ... I think this looks reasonable if you would want to stick with it ... but if you wanna play with it a bit more ... what if you:
* Make the `ApiCredentials` non-static
* Create it in the `fromCrd` method of the CruiseControl class
* Move there the `userManagedApiSecretName` and `userManagedApiSecretKey` fields and the validation methods
* Have a one public field `apiCredential()` in the CruiseControl class to return the ApiCredentials insztance and call things through it: e.g. `cruiseControl.apiCredetnails().generateSecret(...)`

This way, you would better separate the code and the handling of this from the main `CruiseControl` model class, make it smaller and leaner.